docs/date.md

```diff
@@ -1,96 +1,398 @@
 
-```js
-const date = new Temporal.Date(2019, 11, 18); // Monday
-date.dayOfWeek; // 1
+Usage example:
+```javascript
+date = Temporal.Date.from('2006-08-24');
+['MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT', 'SUN'][date.dayOfWeek - 1]  // => THU
+```
+
+### date.**dayOfYear** : number
+
+The `dayOfYear` read-only property gives the ordinal day of the year that the date falls on.
+This is a value between 1 and 365, or 366 in a leap year.
+
+Usage example:
+```javascript
+date = Temporal.Date.from('2006-08-24');
+// Print date in ISO ordinal date format
+year = `${date.year}`.padStart(4, '0');
+day = `${date.dayOfYear}`.padStart(3, '0');
+`${year}-${day}`  // => 2006-236
 ```
 
 ### date.**weekOfYear** : number
 
-Returns the week in a calendar year
+The `weekOfYear` read-only property gives the ISO week number of the date.
+This is normally a value between 1 and 52, but in a few cases it can be 53 as well.
+ISO week 1 is the week containing the first Thursday of the year.
+For more information on ISO week numbers, see for example the Wikipedia article on [ISO week date](https://en.wikipedia.org/wiki/ISO_week_date).
+
+Usage example:
+```javascript
+date = Temporal.Date.from('2006-08-24');
+// Print date in ISO week number / week day format
+year = `${date.year}`.padStart(4, '0');
+week = `${date.weekOfYear}`.padStart(2, '0');
+`${year}-W${week}-${date.dayOfWeek}`  // => 2006-W34-4
+```
 
 ### date.**daysInMonth** : number
 
-Returns the number of days in the month the `Date` Object represents
+The `daysInMonth` read-only property gives the number of days in the month that the date falls in.
+This is 28, 29, 30, or 31, depending on the month and whether the year is a leap year.
+
+Usage example:
+```javascript
+// Attempt to write some mnemonic poetry
+const monthsByDays = {};
+for (let month = 1; month <= 12; month++) {
+    const date = Temporal.now.date().with({month});
+    monthsByDays[date.daysInMonth] = (monthsByDays[date.daysInMonth] || []).concat(date);
+}
+
+const strings = monthsByDays[30].map(date => date.toLocaleString('en', {month: 'long'}));
+// Shuffle to improve poem as determined empirically
+strings.unshift(strings.pop());
+const last = strings.pop();
+const poem = `Thirty days hath ${strings.join(', ')}, and ${last}`;
+
+console.log(poem);
+```
 
 ### date.**daysInYear** : number
 
-Returns the number of days in the year the `Date` Object represents
+The `daysInYear` read-only property gives the number of days in the year that the date falls in.
+This is 365 or 366, depending on whether the year is a leap year.
+
+Usage example:
+```javascript
+date = Temporal.now.date();
+`The year is ${Math.round(date.dayOfYear / date.daysInYear * 100)}% over!`
+// example output: "The year is 10% over!"
+```
 
 ### date.**leapYear** : boolean
 
-Returns a boolean indicating whether the `Date` is in a leap year or not
+The `leapYear` read-only property tells whether the year that the date falls in is a leap year or not.
```

Comment:
Should this be named `isLeapYear` instead, to emphasize that it is a yes or no answer?

Reply:
Opened https://github.com/tc39/proposal-temporal/issues/369 for this.